expertsystem/io/_pdg.py

```diff
@@ -0,0 +1,143 @@
+        pid=int(pdg_particle.pdgid),
+        mass=convert_mass_width(pdg_particle.mass),
+        width=convert_mass_width(pdg_particle.width),
+        state=QuantumState[float](
+            charge=int(pdg_particle.charge),
+            spin=float(pdg_particle.J),
+            strangeness=quark_numbers[0],
+            charmness=quark_numbers[1],
+            bottomness=quark_numbers[2],
+            topness=quark_numbers[3],
+            baryon_number=__compute_baryonnumber(pdg_particle),
+            electron_lepton_number=lepton_numbers[0],
+            muon_lepton_number=lepton_numbers[1],
+            tau_lepton_number=lepton_numbers[2],
+            isospin=__create_isospin(pdg_particle),
+            parity=parity,
+            c_parity=__create_parity(pdg_particle.C),
+            g_parity=__create_parity(pdg_particle.G),
+        ),
+    )
+
+
+__remove_from_quark_content = {"sqrt", "Maybe"}
+
+
+def __compute_quark_numbers(
+    pdg_particle: PdgDatabase,
+) -> Tuple[int, int, int, int]:
+    strangeness = 0
+    charmness = 0
+    bottomness = 0
+    topness = 0
+    if pdg_particle.pdgid.is_hadron:
+        quark_content = pdg_particle.quarks
+        for word in __remove_from_quark_content:
+            quark_content = quark_content.replace(word, "")
+        strangeness = quark_content.count("S") - quark_content.count("s")
+        charmness = quark_content.count("c") - quark_content.count("C")
+        bottomness = quark_content.count("B") - quark_content.count("b")
+        topness = quark_content.count("t") - quark_content.count("T")
+    return (
+        strangeness,
+        charmness,
+        bottomness,
+        topness,
+    )
+
+
+def __compute_lepton_numbers(
+    pdg_particle: PdgDatabase,
+) -> Tuple[int, int, int]:
+    electron_lepton_number = 0
+    muon_lepton_number = 0
+    tau_lepton_number = 0
+    if pdg_particle.pdgid.is_lepton:
+        lepton_number = int(sign(pdg_particle.pdgid))
+        if "e" in pdg_particle.name:
+            electron_lepton_number = lepton_number
+        elif "mu" in pdg_particle.name:
+            muon_lepton_number = lepton_number
+        elif "tau" in pdg_particle.name:
+            tau_lepton_number = lepton_number
+    return electron_lepton_number, muon_lepton_number, tau_lepton_number
+
+
+def __compute_baryonnumber(pdg_particle: PdgDatabase) -> int:
+    return int(sign(pdg_particle.pdgid) * pdg_particle.pdgid.is_baryon)
+
+
+def __create_isospin(pdg_particle: PdgDatabase) -> Optional[Spin]:
+    if pdg_particle.I is None:
+        return None
+    return Spin(pdg_particle.I, __compute_isospin_projection(pdg_particle))
+
+
+def __compute_isospin_projection(pdg_particle: PdgDatabase) -> float:
+    spin_projection = 0.0
+    if pdg_particle.pdgid.is_hadron:
+        for quark in pdg_particle.quarks:
+            if quark in ["u", "D"]:
+                spin_projection += 0.5
+            elif quark in ["U", "d"]:
+                spin_projection -= 0.5
```

Comment:
Ok there is a problem here, originating from the `particle` package side. For example the particles `a0(980)0/+/-` all have a quark content defined as `'Maybe non-qQ'`. So in other words its not set... In my honest opinion, this make the `particle` package pretty untrustworthy. As a workaround, you could use the Gellmann Nishijima formula to calculate the correct isospin projection. https://en.wikipedia.org/wiki/Gell-Mann%E2%80%93Nishijima_formula. @redeboer is already writing a test to check the consistency of particle definitions. This will also use this formula, so you could try to reuse that.

Reply:
@spflueger There are valid reasons to leave the quark content vague or `"Maybe non-qQ"`, see
https://pdg.lbl.gov/2020/reviews/rpp2020-rev-non-qqbar-mesons.pdf

For now, I use a mapping to 'fix' the isospin projections
https://github.com/ComPWA/expertsystem/blob/da271944af63ddd209198a463937b4c814a7fa8f/expertsystem/io/_pdg.py#L138-L146
It's just to sketch the idea, because there are 12 particles with `"Maybe non-qQ"`, see
```python
particle.Particle.findall(lambda p: "qq" in p.quarks.lower())
```

Other unconventional quark content strings are now parsed with regex
https://github.com/ComPWA/expertsystem/blob/da271944af63ddd209198a463937b4c814a7fa8f/expertsystem/io/_pdg.py#L165-L169

I first want to reorganise the `test_particle_collection` tests a bit in another PR, meanwhile we can think whether there is a more stable solution to the `"Maybe non-qQ"` problem.

Reply:
For the record, it concerns these:
- [`f(0)(500)`](https://pdg.lbl.gov/2020/listings/rpp2020-list-f0-500.pdf)
- [`f(0)(980)`](https://pdg.lbl.gov/2020/listings/rpp2020-list-f0-980.pdf)
- [`a(0)(980)0+-`](https://pdg.lbl.gov/2020/listings/rpp2020-list-a0-980.pdf)
- [`pi(1)(1400)0+-`](https://pdg.lbl.gov/2020/listings/rpp2020-list-pi1-1400.pdf)
- [`f(0)(1500)`](https://pdg.lbl.gov/2020/listings/rpp2020-list-f0-1500.pdf)
- [`pi(1)(1600)0+-`](https://pdg.lbl.gov/2020/listings/rpp2020-list-pi1-1600.pdf)

Note that the 𝜋₁ states are exotic (𝐽^𝐶𝑃 = 1⁻⁺), so I guess we have to exclude these from the import.

Reply:
@redeboer Thanks for the additional infos. I'm aware of the reason behind "maybe non-qqbar". However all of the conserved quantum numbers in reactions can precisely be specified, regardless of the precise quark content. So I would make more sense if the `particle` package would provide this info.

Regarding the 𝜋₁ state. "Exotic" states are particularly interesting to handle in reaction, so they should be definitely included as well. Just to clarify, those states are just not "constructable" in the simple meson (quark antiquark) picture. But they are perfectly fine to use inside the expertsystem itself

Reply:
Should be solved with the latest commits. Last thing I want to tackle cleanly is the mismatch in floating point precision (some values from the `particle` package come in as `...0000002`, for instance).